File: src/main/java/com/patres/languagepopup/Point.java
package com.patres.languagepopup;


import com.patres.languagepopup.excpetion.PointFormatException;

public class Point {

    private int x;
    private int y;

    public Point(int x, int y) {
        this.x = x;
        this.y = y;
    }

    public static Point stringToPoint(String pointString) throws PointFormatException {
        try {
            String pointArray[] = pointString.substring(1, pointString.length() - 1).split(";");
            int x = new Integer(pointArray[0]);
            int y = new Integer(pointArray[1]);
            return new Point(x, y);
        } catch (Exception e) {
            throw new PointFormatException(pointString);
        }
    }

    @Override
    public String toString() {
        return String.format("(%d;%d)", x, y);
    }

    public int getX() {
        return x;
    }

    public void setX(int x) {
        this.x = x;
    }

    public int getY() {
        return y;
    }

    public void setY(int y) {
        this.y = y;
    }
}

File: src/main/java/com/patres/languagepopup/Point.kt
package com.patres.languagepopup


import com.patres.languagepopup.excpetion.PointFormatException

class Point(
        var x: Int,
        var y: Int
) {

    override fun toString(): String {
        return String.format("(%d;%d)", x, y)
    }

    companion object {
        fun stringToPoint(pointString: String): Point {
            try {
                val pointArray = pointString.substring(1, pointString.length - 1).split(";".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                val xInt = pointArray[0].toInt()
                val yInt = pointArray[1].toInt()
                return Point(xInt, yInt)
            } catch (e: Exception) {
                throw PointFormatException(pointString)
            }
        }
    }
}
